packages/internal/dex/sdk/src/lib/transactionUtils/approval.ts

```diff
@@ -157,7 +157,7 @@ export const getApproval = async (
     preparedApproval.amount.token.address,
   );
 
-  const gasFeeEstimate = gasPrice ? calculateGasFee(gasPrice, gasEstimate) : null;
+  const gasFeeEstimate = gasPrice ? calculateGasFee(false, gasPrice, gasEstimate) : null;
```

Comment:
Boolean arguments are not considered very ergonomic. Looking at this line in isolation, without context I would have no idea what `false` does or means. Some alternatives:

* Use a separate function e.g. have both `calculateGasFee` and `calculateGasFeeViaSwapProxy()`
* Use an object for options e.g. `calculateGasFee(gasPrice, gasEstimate, { usesSwapProxy: true })`

Both of these help the reader understand what's happening.